store/postgres/src/relational_queries.rs

```diff
@@ -1971,6 +2006,54 @@ impl<'a> Query for FindQuery<'a> {
 
 impl<'a, Conn> RunQueryDsl<Conn> for FindQuery<'a> {}
 
+#[derive(Debug, Clone)]
+pub struct FindRangeQuery<'a> {
```

Comment:
Have you thought about doing it the way `Layout::find` now does it? That would save a roundtrip through JSON. For the block, you'd want to add either `block$` or `lower(block_range) as block$` into the selected columns.

Reply:
It crossed my mind a couple times before but as discussed offline, I would prefer to postpone it for a later moment in order not to increase the complexity here.